storage/remote/queue_manager.go

```diff
@@ -581,6 +581,12 @@ func (t *QueueManager) calculateDesiredShards() int {
 	}
 
 	numShards := int(math.Ceil(desiredShards))
+	// Do not downshard if we are more than ten seconds back.
+	if numShards < t.numShards && samplesPending > 10.0*samplesInRate {
```

Comment:
should use `shardUpdateDuration` here imo

Reply:
I am not sure, they happen to be the same right now, but I would be perfectly happy to change this value to something else that doesn't match the `shardUpdateDuration`. In other words, I don't think they are actually coupled.